### DISCUSSION TITLE
Best Method for Evauating CoP / SCoP for HA

### DISCUSSION THREAD

**Discussion by LAScotland** [signalkraft/mypyllant-component#96] [General]
Hey All,

Has anyone managed to come up with an elegant method for doing this in HA?

Here's my helper template for Live CoP:

```
{{
    (
        ((states('sensor.****_device_0_arotherm_plus_earned_environment_energy_heating')|float)+(states('sensor.****_device_0_arotherm_plus_earned_environment_energy_domestic_hot_water')|float))
       /
        ((states('sensor.****_device_0_arotherm_plus_consumed_electrical_energy_heating')|float)+(states('sensor.****_device_0_arotherm_plus_consumed_electrical_energy_domestic_hot_water')|float))
    )
    *100
}}
```

Which works, obviously, but integrating under the curve of those daily resetting sensors and then returning a SCoP figure is a bit harder. Any ideas?

**Comment by willenic** [comment]
Hi, you could use 2 [utility meters ](https://www.home-assistant.io/integrations/utility_meter/). One for yield one for consumption. Then use a template to calculate the SCoP.
